From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andras Becsi <andras.becsi@digia.com>
Date: Fri, 14 Mar 2014 18:37:44 +0100
Subject: Cherry-pick https://codereview.chromium.org/137433002 from upstream

This fixes the build with a cross compiler toolchain.
We do not require the functionality of this script but
gyp tries to run it and fails if the specified file
does not exist.

As the upstream issue mentions, this script is "going
away soonish", until then we can keep this patch around.

Change-Id: Ib9084103cd95aad3c48541e13351ebee2c9279ea
Reviewed-by: Pierre Rossi <pierre.rossi@gmail.com>
---
 build/linux/python_arch.sh | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/build/linux/python_arch.sh b/build/linux/python_arch.sh
index 3a41f94..e0f63bc 100755
--- a/build/linux/python_arch.sh
+++ b/build/linux/python_arch.sh
@@ -11,7 +11,9 @@
 #
 
 file_out=$(file --dereference "$1")
-if [ $? -ne 0 ]; then
+# The POSIX spec says that `file` should not exit(1) if the file does not
+# exist, so do our own -e check to catch things.
+if [ $? -ne 0 ] || [ ! -e "$1" ] ; then
   echo unknown
   exit 0
 fi
